File: Black_Scholes_Call_Put_Option_Price/C#/Black_Scholes_Call_Put_Option.cs
using System;
using MathNet.Numerics;

/// <summary>
/// This class computes Black-Scholes Call and Put Option Prices getting parameters from the user.
/// </summary>
class Black_Scholes_Call_Put_Option
{
    /// <summary>
    /// This method computes Black-Scholes Call and Put Option Prices.
    /// </summary>
    /// <param name="S">Stock Price</param>
    /// <param name="K">Strike</param>
    /// <param name="r">Interest Rate</param>
    /// <param name="T">Time to Maturity</param>
    /// <param name="sigma">Volatility</param>
    /// <returns>
    /// <c><paramref name="Call"/></c> is Black-Scholes Call Option Price.
    /// <c><paramref name="Put"/></c> is Black-Scholes Put Option Price.
    /// </returns>
    static (double Call, double Put) Black_Scholes_Call_Put_Option_Price(double S, double K, double r, double T, double sigma)
    {
        if (T == 0)
        {
            double Call = Math.Max(S - K, 0.0);
            double Put = Math.Max(K - S, 0.0);
            return (Call, Put);
        }
        else
        {
            double d1 = (Math.Log(S / K) + (r + Math.Pow(sigma, 2) / 2) * T) / (sigma * Math.Sqrt(T));
            double d2 = (Math.Log(S / K) + (r - Math.Pow(sigma, 2) / 2) * T) / (sigma * Math.Sqrt(T));
            double Call = S * N(d1) - Math.Exp(-r * T) * K * N(d2);
            double Put = Math.Exp(-r * T) * K * N(-d2) - S * N(-d1);
            return (Call, Put);
        }
    }

    /// <summary>
    /// This method computes Normal (Gaussian) CDF.
    /// </summary>
    static double N (double x) {
        MathNet.Numerics.Distributions.Normal result = new MathNet.Numerics.Distributions.Normal ();
        return result.CumulativeDistribution (x);
    }

    static void Main()
    {
        double Call, Put;
        double S, K, r, T, sigma;

        Console.WriteLine("Stock Price:");
        S = Convert.ToDouble(Console.ReadLine());
        Console.WriteLine("Strike:");
        K = Convert.ToDouble(Console.ReadLine());
        Console.WriteLine("Interest Rate:");
        r = Convert.ToDouble(Console.ReadLine());
        Console.WriteLine("Time to Maturity:");
        T = Convert.ToDouble(Console.ReadLine());
        Console.WriteLine("Volatility:");
        sigma = Convert.ToDouble(Console.ReadLine());

        (Call, Put) = Black_Scholes_Call_Put_Option_Price(S, K, r, T, sigma);
        Console.WriteLine("Call==" + Call);
        Console.WriteLine("Put==" + Put);
    }
}

File: Black_Scholes_Call_Put_Option_Price/Java (Gradle (Groovy))/BlackScholes/src/main/java/BlackScholesCallPutOption.java
import java.util.Scanner;
import org.apache.commons.math3.distribution.NormalDistribution;

public class BlackScholesCallPutOption {

    /**
     * This method computes Black-Scholes Call and Put Option Prices.
     *
     * @param S      Stock Price
     * @param K      Strike
     * @param r      Interest Rate
     * @param T      Time to Maturity
     * @param sigma  Volatility
     * @return an array containing Call and Put option prices
     */
    static double[] blackScholesCallPutOptionPrice(double S, double K, double r, double T, double sigma) {
        double[] result = new double[2]; // result[0] = Call, result[1] = Put

        if (T == 0) {
            result[0] = Math.max(S - K, 0.0);
            result[1] = Math.max(K - S, 0.0);
        } else {
            double d1 = (Math.log(S / K) + (r + Math.pow(sigma, 2) / 2) * T) / (sigma * Math.sqrt(T));
            double d2 = (Math.log(S / K) + (r - Math.pow(sigma, 2) / 2) * T) / (sigma * Math.sqrt(T));

            NormalDistribution normalDistribution = new NormalDistribution();

            result[0] = S * normalDistribution.cumulativeProbability(d1) - Math.exp(-r * T) * K * normalDistribution.cumulativeProbability(d2);
            result[1] = Math.exp(-r * T) * K * normalDistribution.cumulativeProbability(-d2) - S * normalDistribution.cumulativeProbability(-d1);
        }
        return result;
    }


    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        double S, K, r, T, sigma;

        System.out.println("Stock Price:");
        S = Double.parseDouble(scanner.nextLine());
        System.out.println("Strike:");
        K = Double.parseDouble(scanner.nextLine());
        System.out.println("Interest Rate:");
        r = Double.parseDouble(scanner.nextLine());
        System.out.println("Time to Maturity:");
        T = Double.parseDouble(scanner.nextLine());
        System.out.println("Volatility:");
        sigma = Double.parseDouble(scanner.nextLine());

        double[] result = blackScholesCallPutOptionPrice(S, K, r, T, sigma);
        double Call = result[0];
        double Put = result[1];

        System.out.println("Call = " + Call);
        System.out.println("Put = " + Put);
    }

}
